ta/pkcs11/src/processing.c

```diff
@@ -195,8 +197,35 @@ static enum pkcs11_rc generate_random_key_value(struct obj_attrs **head)
 	TEE_GenerateRandom(value, value_len);
 
 	rc = add_attribute(head, PKCS11_CKA_VALUE, value, value_len);
+	if (rc != PKCS11_CKR_OK)
+		goto out;
+
+	/* Remove the default empty check value attribute if found */
+	rc = remove_empty_attribute(head, PKCS11_CKA_CHECK_VALUE);
```

Comment:
Ditto about not removing the attribute if present.

Reply:
The template passed to C_GenerateKey can set CKA_CHECK_VALUE attribute. In which case we should not omit it but ensure it complies with a valid value, that is either a zero-sized value or the value the pkcs11 TA also computes. Other values for this attribute should error with return code CKR_ATTRIBUTE_VALUE_INVALID.